src/xSTIR/pSTIR/STIR.py

```diff
@@ -1441,49 +1453,64 @@ def set_acquisition_sensitivity(self, asm):
         norm:  an AcquisitionSensitivityModel
         object containing normalisation n.
         """
+        if self.const:
+            msg = 'cannot set_acquisition_sensitivity'
+            raise RuntimeError(msg + ' for a const object')
         assert_validity(asm, AcquisitionSensitivityModel)
         parms.set_parameter(
             self.handle, 'AcquisitionModel', 'asm', asm.handle)
         # save reference to the Acquisition Sensitivity Model
         self.asm = asm
 
-    def forward(self, image, subset_num=0, num_subsets=1, ad=None):
+    def forward(self, image, subset_num=0, num_subsets=1, out=None):
         """Return the forward projection of image.
 
         image   :  an ImageData object.
         """
         assert_validity(image, ImageData)
-        if ad is None:
+        if out is None:
             ad = AcquisitionData()
-            ad.handle = pystir.cSTIR_acquisitionModelFwd(
-                self.handle, image.handle, subset_num, num_subsets)
+            if self.at is None and self.bt is None:
```

Comment:
no, we have to call `cSTIR_acquisitionModelLinFwd` here in case `self` is obtained from non-linear (affine) model via `get_linear_acquisition_model`

Reply:
I haven't checked all this properly, but I thought that in that case, `at` and `bt` will be "zero", and the normal `forward` already handles that graciously.

Reply:
`get_linear_acquisition_model` creates a const reference to generally non-linear model, so `sptr_add_` and/or `sptr_background_` may point to non-zero data

perhaps it would be less confusing to change this line to `if self.is_linear()`?

Reply:
hmm. that's a bit scary. so this is a Python-only fix, where the underlying C++ object doesn't know what it is? I don't really like that I'm afraid.

Reply:
this is just another quick fix not to hold back CIL people

just like with `DataContainer`'s `power`, `maximum` etc. etc. , which must eventually move to C++

Reply:
yeah. I'm getting a bit concerned about the amount of quick fixes I guess... fine for now.